### DISCUSSION TITLE
I'm stucked on authenticating with oauth Google

### DISCUSSION THREAD

**Discussion by adellamaggiora** [supabase/supabase#16495] [Questions] [answered]
I haven't been able to figure out a problem for days:
I'm using an Angular project and therefore the Supabase.js library for client management.

After authenticating with Google's Oauth, the provider Google redirects to my application, providing access_token and refresh_token in the url.
If then checked on the site [](https://jwt.io/ )and it appears to be a valid token.
Also, if I enter my jwtkey the signature is verified.

The main problem is that, despite this, Supabase doesn't set up the session, I checked the local storage and there is nothing.
When instead I authenticate with email the session is set up correctly.

How can I handle this thing?

I thank all those who will want to help me.

**Comment by silentworks** [comment]
Is this a normal Angular client side app or is it a Angular Universal app? Universal apps might require more setup to get working as they include both client and server side code. A client only app should work as is, you can checkout the user-management Angular example to see how it's setup https://github.com/supabase/supabase/tree/master/examples/user-management/angular-user-management

**Reply by adellamaggiora** [reply]
Is an Angular client side app. 
I am using capacitor with ionic but for now I' just developing the web version, I'll figure out the native auth with google in a second moment. For now I'd just want to work in a browser environment.
In the example I cannot see a Google login, just magic link.
The authentication through email works fine to me too. 
The problem is that when google redirects to my app the user session is not setted so I can't call my API and I result as not authenticated.

**Comment by jayleaton** [comment]
I have the same problem in my next project the flow works correctly with email and password singin but oauth causes a weird state that forces the user to refresh. it recognises that the user logs in but doesnt setup auth state

**Reply by adellamaggiora** [reply]
Also if I check inside the table auth.user everything is ok, the column "last_sign_in_at" onto the db is correctly updated and the provider is Google. Is it a Supabase issue? I can't figure out what's going on.

**Reply by jayleaton** [reply]
my issue turned out to be a client side error I used the `onAuthStateChange` helper to force a refresh I am using next so not sure if this is the same issue or not now

**Reply by adellamaggiora** [reply]
I am using `onAuthStateChange` too but the function is not triggered.

**Comment by adellamaggiora** [comment, marked as answer]
I confirm it was a Supabase issue. 
I faced the issue with client supabase-js v. 2.29.0.
After the installation of the new client (v. 2.33.1 using the same application code) the session is automatically setted and everything works fine.
